### PR TITLE
Stian sonarqube cleanup

### PR DESCRIPTION
**Pull request by stian-sandvold**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java [hunk 1]
```diff
@@ -305,48 +303,33 @@ public App installApp( File file, String filename )
...
 
-                    }
-                    catch ( IOException e )
-                    {
-                        e.printStackTrace();
-                    }
+                }
+                catch ( IOException e )
+                {
+                    e.printStackTrace();
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java]: printStackTrace() should be discouraged right?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java [hunk 2]
```diff
@@ -261,14 +261,12 @@ public App installApp( File file, String filename )
         App app = new App();
         log.info( "Installing new app: " + filename );
         
-        try
+        try( ZipFile zip = new ZipFile( file ) )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java]: Remember to actually test this part well ;)

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultJobConfigurationService.java [hunk 3]
```diff
@@ -153,7 +154,7 @@ public JobConfiguration getJobConfiguration( int jobId )
             for ( Field field : clazz.getDeclaredFields() )
             {
                 if ( Arrays.stream( field.getAnnotations() )
-                    .anyMatch( f -> f.annotationType().getSimpleName().equals( "JsonProperty" ) ) )
+                    .anyMatch( f -> f instanceof JsonProperty ) )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultJobConfigurationService.java]: Remember to test well ;)

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/TableAlteror.java [hunk 4]
```diff
@@ -1569,33 +1569,30 @@ private void upgradeImplicitAverageMonitoringRules()
...
-                List<Integer> foreignIds = new ArrayList<>();
+        for ( Integer distinctId : distinctIds )
+        {
+            List<Integer> foreignIds = new ArrayList<>();
 
-                ResultSet resultSet = statement.executeQuery( "SELECT " + col2 + " FROM " + table + " WHERE " + col1
-                    + "=" + distinctId );
+            String query = "SELECT " + col2 + " FROM " + table + " WHERE " + col1 + "=" + distinctId;
 
+            try (ResultSet resultSet = statement.executeQuery( query ))
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/TableAlteror.java]: Codestyle, spaces.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/TableAlteror.java [hunk 5]
```diff
@@ -1569,33 +1569,30 @@ private void upgradeImplicitAverageMonitoringRules()
...
-        catch ( Exception ex )
-        {
-            log.error( ex );
-        }
-        finally
-        {
-            holder.close();
-        }
+
+        holder.close();
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/TableAlteror.java]: I'm not sure if we should do this, perhaps revert, we need to ensure that "close()" is always called. My old StatementHolder stuff I think needs itself to implement Closeable as the holder.close() operation is closing the underlying streams.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/startup/TableAlteror.java [hunk 6]
```diff
@@ -283,12 +283,12 @@ private void updateTrackedEntityInstanceGeometry()
...
-        {
-            Statement statement = holder.getStatement();
+        Statement statement = holder.getStatement();
 
-            ResultSet resultSet = statement
-                .executeQuery( "SELECT count(column_name) > 0 FROM information_schema.columns WHERE table_name='trackedentityinstance' AND column_name='featuretype'" );
+        String query = "SELECT count(column_name) > 0 FROM information_schema.columns WHERE table_name='trackedentityinstance' AND column_name='featuretype'";
+
+        try( ResultSet resultSet = statement.executeQuery( query ) )
+        {
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/startup/TableAlteror.java]: See above.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java (hunk 2 above)
Comment (stian-sandvold) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java]: tested

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultJobConfigurationService.java (hunk 3 above)
Comment (stian-sandvold) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultJobConfigurationService.java]: tested

**Pull request merged**